Feature/6.x/object addons

eric-lamb62 opened the pull request:
EE6 version of https://github.com/ExpressionEngine/ExpressionEngine/pull/2367

Docs: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/503

(x.X.x) Small additive changes should target the next minor branch (release/next-minor if a numbered branch does not yet exist)

<!-- What's in this pull request? -->
## Overview

This PR allows for Add-on developers to create their Add-on methods using an abstracted object layer while keeping backwards compatibility intact. It covers Extensions, Modules (Action and Tags), and the Control Panel layers. It's very simple to implement (just inherit an object), while maintaining the legacy implementation in play. 

To implement, you simply set your Add-on `mod.`, `ext.`, and/or `mcp` files to inherit from a base object (depending on the implementation).

## Add-on File Examples

Here's a breakdown for how the Add-on files would look for an Add-on named `Custom_addon`.

> Note that this works by using set or determined namespaces for a given Add-on. By default, this Controller implementation will use the Add-on name to get the configured namespace within the Add-on. For this example of `Custom_addon`, it's `Mithra62\PlayThing`; all route namespaces are determined by that. 

### Module

The `mod` file would consist exclusively of the below. 

```php
use ExpressionEngine\Service\Addon\Module;

class Custom_addon extends Module
{
    protected $addon_name = 'custom_addon';
}
```

Once that's done, your Add-on Module is now set to look for objects for all calls to it. Those objects depend on what "type" (Tag or Action). Using the configured namespace for the Add-on, the Module routes would be looked for in `Mithra62\PlayThing\Module\Tags|Action`. 

#### Template Tag 

For a template tag called as `{exp:custom_addon:test_template_tag}`, you'd setup your Tag object like the below:

```php
namespace Mithra62\PlayThing\Module\Tags;

use ExpressionEngine\Service\Addon\Controllers\Tag\AbstractRoute;

class TestTemplateTag extends AbstractRoute
{
    public function process()
    {

    }
}
```

#### Actions
Just like with Template Tags, Actions get stored in the `Actions` namespace. The below is for an Action called `my-test-action`. 

```php
<?php
namespace Mithra62\PlayThing\Module\Actions;

use ExpressionEngine\Service\Addon\Controllers\Action\AbstractRoute;

class MyTestAction extends AbstractRoute
{
    public function process()
    {

    }
}
```

### Extension

The `ext` is structured just like the others:

```php
use ExpressionEngine\Service\Addon\Extension;

class Custom_addon_ext extends Extension
{
    protected $addon_name = 'custom_addon';
}
```

Again, just like the others, Extensions are structured similarly.

```php
<?php
namespace Mithra62\PlayThing\Extensions;

use ExpressionEngine\Service\Addon\Controllers\Extension\AbstractRoute;

class TestExtension extends AbstractRoute
{
    public function process()
    {

    }
}
```

Note that the above `process` method allows you to accept variable parameters (just like regular Extensions). For example, to build an Extension around `template_post_parse`, you'd write your Extension like the below:

```php
namespace Mithra62\PlayThing\Extensions;

use ExpressionEngine\Service\Addon\Controllers\Extension\AbstractRoute;

class TestExtension extends AbstractRoute
{
    public function process(string $final_template, bool $is_partial, string $site_id, array $currentTemplateInfo): string
    {
        return $final_template;
    }
}
```

### Module Control Panel (MCP)

The `mcp` layer acts more as a Controller/Router than the others in that the CP is supposed to be rendered as Views. To start, just like all the others, you extend from the `Mcp` object. Once that's done, all requests will be routed to the object layer. 

```php
use ExpressionEngine\Service\Addon\Mcp;

class Custom_addon_mcp extends Mcp
{
    protected $addon_name = 'custom_addon';
}
```

#### Route Examples

A basic 'index' route (normally done through a method attached to the `mcp` object) would look like the below:

```php
namespace Mithra62\PlayThing\Mcp;

use ExpressionEngine\Service\Addon\Controllers\Mcp\AbstractRoute;

class Index extends AbstractRoute
{
    /**
     * @var string
     */
    protected $route_path = 'index';

    /**
     * @var string
     */
    protected $cp_page_title = 'home';

    /**
     * @param false $id
     * @return AbstractRoute
     */
    public function process($id = false): AbstractRoute
    {
        return $this;
    }
}
```

The generation of the CP array format is done automatically by the Controller so all the devs really do is assign variables and ensure their `process` method returns an instance of their Route:

```php
namespace Mithra62\PlayThing\Mcp;

use ExpressionEngine\Service\Addon\Controllers\Mcp\AbstractRoute;

class Index extends AbstractRoute
{
    /**
     * @var string
     */
    protected $route_path = 'index';

    /**
     * @var string
     */
    protected $cp_page_title = 'home';

    /**
     * @param false $id
     * @return AbstractRoute
     */
    public function process($id = false): AbstractRoute
    {
        $this->addBreadcrumb('test-link', 'test-link')
             ->addBreadcrumb('test-link2', 'test-link2');

        $variables = [];
        $this->setBody('my-view', $variables);
        return $this;
    }
}
```

## Nature of This Change

<!-- Check all that apply: -->

- [x] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [x] 🛁 Refactors existing code
- [x] 💅 Fixes coding style
- [x] ✅ Adds tests
- [x] 💥 Improves Maintainability
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security <!-- if your fix would EXPOSE a current security flaw, do not submit a pull request. Instead report a security bug at https://docs.expressionengine.com/latest/bugs_and_security_reports -->

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation
<!-- Required for new features -->
User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine/pull/2110

<!-- Don't forget to add a single line changelog for your change to the appropriate file!

- changelogs/patch.rst (x.x.X)
- changelogs/minor.rst (x.X.x)
- changelogs/major.rst (X.x.x)

Thank you for contributing to ExpressionEngine! -->

intoeetive committed:
add `edit_date` column in Entry Manager

EECORE-1788

fixes #1850

(cherry picked from commit d93767d68f2acb2e3414dc1907994888864d0fb9)

Yulyaswan committed:
Updated styles for the "Edit Category Group" tab "Fields"

EECORE-1819

Yulyaswan committed:
Fixed styling of Create/Edit Category Field > Select Dropdown

(cherry picked from commit 4b5eff6c0bc46d324381a8753c30bd43d2d54920)

danielpunkass committed:
Don't filter the getRecentPosts query on channel when we've been provided a specific post ID to fetch. This fixes a problem that prevented getPost from working as expected. Without this fix the getPost call would always return an empty string unless the default channel ID happened to be the same as the channel for the post ID being requested.

Yulyaswan committed:
Fixed empty state styling on main Categories screen

EECORE-1062

Commit:
Adding extension hook to be executed upon cache clearing

Commit:
Adding changelog line

eric-lamb62 commented:
I do feel there's an argument to be made to switch the namespace(s) up to `\ExpressionEngine\Controllers\Addon\*` but with the `Installer` object being where it is, 'twas a little confusing on which was ideal.

jcogs-design committed:
Fix php8.1 null assignment deprecation warning

Line 1779 is passing null to `str_replace` under some conditions and assuming null will be replaced by ''.  php 8.1 doesn't like this and issues a deprecation warning. This uses the null coalescing operator to replace null with '' when the function is called if necessary.

robinsowell committed:
If an add-on url is wrong it was throwing a php error

Better to throw a 404 than show an error (which was confusing).

We were getting a 'Call to a member function checkCachedLicenseResponse() on null' in Controller/Addons/Addons.php line 745 when we had a malformed add-on url.  Tossed the other 404s in as well since they just made sense to me, but probably not needed.

(cherry picked from commit 1b52fe5ec74a8f20dd8e30619d07ce2de4d8c9a6)

jHards committed:
Handle form_class in form_declaration tag within forgot_form template

intoeetive committed:
fix the File Usage utility to correctly update Fluid fields

fix to keep the total_usage correct on subsequent runs

EECORE-2072

intoeetive committed:
update the regex to properly get file references

EECORE-2071

intoeetive committed:
update version to 7.0.3

intoeetive committed:
fix calculating file usage for files in subfolders

intoeetive committed:
fix parsing the URL for getting image usage stats

intoeetive committed:
do not cache file-by-path check, as that gives an error

intoeetive committed:
Resolved #2208 where language files were for some add-ons were not placed in correct location

intoeetive committed:
Resolved #1561 where file column in file grid field was not searchable

EECORE-1789

intoeetive committed:
fix MySQL error in channel form when field limits by member group instead of role

intoeetive committed:
Improved UI for editing system message templates

EECORE-1792

intoeetive committed:
check the button for data-group-toggle and fix dependent block behavior, EECORE-1678

Yulyaswan committed:
fixed issue when Grid Field within Fluid Field gets cut off

intoeetive committed:
fix the issue where Grid inside Fluid was not empty initially

Yulyaswan committed:
changed sortable to draggable

intoeetive committed:
only add one file usage record to pivot table

EECORE-2075

intoeetive committed:
fix total_records counter on file

intoeetive committed:
make sure pivot record exists only once

matthewjohns0n committed:
Update release date to today

matthewjohns0n committed:
Merge pull request #2225 from ExpressionEngine/release/7.dev/update-release-date

Update release date to today

matthewjohns0n committed:
Merge pull request #2224 from ExpressionEngine/7.dev

ExpressionEngine 7.0.3

intoeetive committed:
correct "folder deleted" message

EECORE-2059

Yulyaswan committed:
fixed fieldset width inside fluid

intoeetive committed:
Resolved #2115 where conditional validation did not work properly on existing objects

EECORE-1935

(cherry picked from commit 5d2f3f741e7a21eef4e893503c7b643051f7f7ad)

Yulyaswan committed:
Added deferred Relationship field option

intoeetive committed:
Resolves #2109 where sometimes loading third-party dashboard widgets could cause PHP error

Yulyaswan committed:
fixed checkbox behaviour in the grid field settings

intoeetive committed:
Resolved #2126 where "add field" button was present for field groups that are not saved yet

intoeetive committed:
Resolved #2136 where formatters for duration field set to hours type were not parsed properly

matthewjohns0n committed:
Merge branch '7.dev' into feature/7.dev/add-structure-to-core

intoeetive committed:
fix setting username for validation on password reset

(cherry picked from commit 89b7a39d5eae8e981b5f6cff06c8fa07ca72d53a)

intoeetive committed:
Resolved #2105 where adding new row for File Grid inside Fluid could cause validation error in some cases

Fixes #2145

EECORE-1899

intoeetive committed:
Resolved #2076, #2141 where button to update template routes was missing some functionality

intoeetive committed:
check directory permissions on new File Grid rows in Fluid

jcogs-design committed:
Update to add php 5.6 compatibility

As per requested change.

jcogs-design committed:
Updated

 with PSR-12 and performance changes requested

intoeetive committed:
Resolved #2127 where there was a typo in defintion of Extension model

(cherry picked from commit cb0f5ec50640187875bcbeb1c8d1a43b1c8fcfd3)

Yulyaswan committed:
fixed style for em in Notes fieldtype

Yulyaswan committed:
fixed issue with wrong value for third column of conditional field v7

matthewjohns0n committed:
Big license/EE update

intoeetive committed:
Resolved #2255 where "Anonymize Consent Logs" feature was not fully functional

Yulyaswan committed:
fixed styles for Color picker field panel, to prevent hidden inside grid

Yulyaswan committed:
Changed the position for dropdown inside relationship field

matthewjohns0n committed:
Remove unused lang file

intoeetive committed:
temporary use old documentation URL

intoeetive committed:
add waiting on Grid test

intoeetive committed:
Release/7.1.0 (#2264)

* bump version to 7.1.0

* little extra waiting on watermark tests

(cherry picked from commit feb7c147813d697d519f406bb8e14d5bb69216e8)

* temporary skip field tests

* set `numTestsKeptInMemory: 0` for tests to avoid memory issues

* Revert "set `numTestsKeptInMemory: 0` for tests to avoid memory issues"

This reverts commit efa7348d3ae5b4d065c7d0a2341d07da3d00bb03.

intoeetive committed:
Merge pull request #2176 from ExpressionEngine/feature/7.dev/add-structure-to-core

Added Structure as first-party add-on, shipped with ExpressionEngine

jcogs-design committed:
Updated to use empty() method

Haha - will remind me to engage brain before doing updates... :)

matthewjohns0n committed:
Merge pull request #2210 from ExpressionEngine/feature/7.x/installed-addons-counter

Added indicator to display number of installed add-ons, #1944

matthewjohns0n committed:
Merge pull request #2216 from ExpressionEngine/feature/7.x/category-ui-fixes

Improved UI for category pages

matthewjohns0n committed:
Merge pull request #2214 from ExpressionEngine/feature/7.x/sort-fields-on-channel-edit

Improved UI for field assignment

matthewjohns0n committed:
Merge pull request #2215 from ExpressionEngine/feature/7.x/system-message-templates-ui

Improved UI for editing system message templates

matthewjohns0n committed:
Add jquery-migrate-3.0.0 library, and include it in the control panel footer

intoeetive committed:
Resolved #1358 where setting CP Homepage for member could cause PHP error on large sites using MSM

intoeetive committed:
Resolved issue when sometimes avatar could not be uploaded on frontend

intoeetive committed:
Resolved #1901 where add-ons with related models were not properly uninstalled

Yulyaswan committed:
check if command btn + click pressed on the same time

Yulyaswan committed:
check if command btn + click pressed on the same time

intoeetive committed:
fix the test template

intoeetive committed:
modify the templates for this test

intoeetive committed:
Merge pull request #2265 from robsonsobral/patch/outdated-resource

Resolved #2063 where outdated resources could be served

intoeetive committed:
Merge pull request #2273 from robsonsobral/feature/Resolved-#1193-where-search-total-results-cannot-have-conditionals-or-no-results

Enable exp:search:total_results to be parsed as tag pair, #1193

intoeetive committed:
Merge pull request #2274 from robsonsobral/feature/has_children

Added has_children variable to category tags, #1726

intoeetive committed:
Merge pull request #2247 from jcogs-design/Issue-2096

Added support for table alias in SQL joins when using Active Record, #2096

intoeetive committed:
Resolved #2263 where the instructions for adding IP to Blocked list were not correct

intoeetive committed:
Resolved #2279 where ColorPicker fields were not initialized on page load

intoeetive committed:
Resolved #2279 where ColorPicker fields were not initialized on page load

intoeetive committed:
Resolved #2135 where categories were not parsed in related categories mode

EECORE-2096

intoeetive committed:
fix loading cache clearing hook on Cypress tests

intoeetive committed:
fix loading cache clearing hook on Cypress tests

intoeetive committed:
Resolved #2122 where Live Preview was not working correctly when template was using search on checkboxes column in Grid

matthewjohns0n committed:
Only count members with CP access

bryannielsen committed:
Update subfolder delete test

matthewjohns0n committed:
Merge pull request #2286 from ExpressionEngine/feature/7.dev/only-count-members-with-cp-access

Only count members with CP access for EE Pro

intoeetive committed:
Resolved issue where PHP notices were shown on Entry Manager page with saved views when migrating from EE6 to EE7

EECORE-2095

intoeetive committed:
Resolved #2285 where exception shown after creating select field populating from other field, but no source field selected

(cherry picked from commit d5b673ea8189fdc3a880ac7cae71ce17d3964075)

intoeetive committed:
add validation for setting of fields populating from other field

(cherry picked from commit 4128b4b03eedaab116b5c15120c2a6686f210c31)

intoeetive committed:
update entry manager that were serialized to json format

bryannielsen committed:
Update default filesystem to use root of system path

EECORE-2093

intoeetive committed:
update Pro license message in dev logs once a week

TomJaeger committed:
Update upgrade_success_banner.php

Changed the last bucket in the upgrade banner

Yulyaswan committed:
cope css file from addon to the core file

intoeetive committed:
ExpressionEngine 7.1.0

bryannielsen committed:
Fix upload writable check to use upload_path

intoeetive committed:
Resolved issue when member role record was not deleted after deleting member

Yulyaswan committed:
fixed styles for structure dark theme

Yulyaswan committed:
add styles

intoeetive committed:
Resolved #2298 where Publish Layouts were not accessible for member via secondary role

Yulyaswan committed:
fixed relationship field if deferred option is on

Yulyaswan committed:
fixed relationship field if deferred option is on

Yulyaswan committed:
fixed relationship dropdown items inside grid

bryannielsen committed:
Update query to check cp_member_count

intoeetive committed:
update version to 7.1.1

intoeetive committed:
Resolved #2297 where deferring relationship field was not working for fields with data

intoeetive committed:
Resolved #2279 where ColorPicker field did not display saved values

intoeetive commented on system/ee/ExpressionEngine/Library/String/Str.php:
```diff
@@ -0,0 +1,43 @@
+    {
+        return self::path2ns(
+            str_replace(' ', '/',
+                str_replace('cartthrob', 'CartThrob', // ;)
+                    ucwords(str_replace('_', ' ', $value))
+                )
```
I think hardcoded `CartThob` probably went in here by mistake?

eric-lamb62 replied:
Yeah, can't provide special consideration to Add-ons; just won't scale lol

Will clean up.

intoeetive reviewed (changes requested):
@mithra62 This looks really solid!

I see you are using type hinting for function return values, - unfortunately we can't take that into EE 6, because it needs to support PHP 5.6, and return type hinting is PHP7+ thing only.

Is that something you can update?

Yulyaswan committed:
added code for ctrl and shift buttons

intoeetive committed:
7.1.2

intoeetive committed:
Merge pull request #2315 from ExpressionEngine/bug/7.x/colorpicker-not-initialized

Resolved #2279 where ColorPicker field did not display saved values

intoeetive committed:
Merge pull request #2314 from ExpressionEngine/bug/7.x/deferred-relationships-EECORE-2102

Resolved #2297 where deferring relationship field was not working for fields with data

Yulyaswan committed:
fixed file upload to almost empty folder

intoeetive committed:
update license check query

intoeetive committed:
Merge branch '7.dev' into bug/7.x/delete-pivot-role-records-when-deleting-member

# Conflicts:
#	build-tools/build.json
#	system/ee/ExpressionEngine/Addons/pro/Service/Access/Access.php
#	system/ee/ExpressionEngine/Tests/bootstrap.php
#	system/ee/installer/controllers/wizard.php
#	system/ee/installer/updates/ud_7_01_01.php
#	system/ee/legacy/libraries/Core.php
#	tests/cypress/support/config/config.php

intoeetive committed:
bring back update file

Yulyaswan committed:
fixed moving between tabs in layout

Yulyaswan committed:
removed Window button code from script

bryannielsen committed:
Properly encode urls created by the filesystem

bryannielsen committed:
Update url encoding to only affect filename

bryannielsen committed:
Change to str_replace for better performance

intoeetive committed:
Merge pull request #2318 from ExpressionEngine/bug/7.x/urlencode-file-url

Properly encode urls created by the filesystem

intoeetive committed:
Merge pull request #2300 from ExpressionEngine/bug/7.x/delete-pivot-role-records-when-deleting-member

Resolved issue when member role record was not deleted after deleting member

intoeetive committed:
fix the button selector for template routes test

intoeetive committed:
add the deferred relationships change that was lost

intoeetive committed:
add the deferred relationships change that was lost

intoeetive committed:
Resolved #2116 where disabling MFA globally could cause redirect loop on login for some users

EECORE-2107

intoeetive committed:
update selector

intoeetive committed:
Merge pull request #2320 from ExpressionEngine/fix/7.x/template-routes-test

fix the button selector for template routes test

eric-lamb62 commented:
Thanks @intoeetive! I built this using the rules we had for the "Cp/Form" Libraries, so focused on 7.1 as the baseline. I'm sure it wouldn't be too difficult to remove the hinting though. Just to double check; you want me to downgrade this to work with PHP 5.6, right?

intoeetive commented:
@mithra62 yes, that is correct.

Or maybe you'll want to change the base for this be 7.dev and make a new one that's compatible with PHP 5.6 to go into 6.dev - that's up to you (unless you think this needs to go into EE7 only, we want to have this included both for 6 and 7 versions of CMS)

eric-lamb62 commented:
Gotcha and agree; it'd be more helpful to have in both 6 and 7. Will clean it up this week(ish).

eric-lamb62 committed:
removes typing

eric-lamb62 commented:
@intoeetive I believe I hit them all but let me know if anything was missed.

intoeetive committed:
Resolved #2279 where ColorPicker dropdown was not visible in new Grid rows

intoeetive commented:
Looks like `system/ee/ExpressionEngine/Library/String/Str.php` wasn't updated

Yulyaswan committed:
added confirm_exit modal to prevent close the relasionship modal window if it has a data

intoeetive committed:
removed class that caused double field initialization

intoeetive committed:
update CKEditor to v35

added CodeBlock button
added ImageCaption button
reorganized image alignment buttons

eric-lamb62 committed:
removes add-on special case and typing

eric-lamb62 commented:
my bad @intoeetive all pushed

Yulyaswan committed:
fixed initialization colorpicker for all field types

intoeetive committed:
Resolved #2331 where saving template revision was causing SQL error when using long table prefix

Yulyaswan committed:
fixed Add new button for Color Picker Swatches block inside grid

intoeetive committed:
support using external stylesheet in RTE using CKEditor

intoeetive committed:
support using external stylesheet in RTE using Redactor

intoeetive committed:
update Redactor to v3.5.2

intoeetive committed:
fix issues with Redactor build

bryannielsen committed:
Fix upload to not follow symlink out of destination

Previously using symlink path in your Upload Destination's server_path
could make it impossible to upload a file

intoeetive reviewed (dismissed):
Looks good to me! Approving

If you could also add some docs for this (even just based on the description of this PR), that would be awesome

intoeetive committed:
Resolved #2335 where having references to non-existing files in File field could case PHP error in Entry Manager listing

intoeetive committed:
add inputs for RTE advanced configuration

Yulyaswan committed:
added space between textarea text and list number

intoeetive committed:
Revert "added space between textarea text and list number"

This reverts commit c17eae2ec4c02a99d7ae9713a0f448a08e624f32.

intoeetive committed:
save RTE config JSON

dynamic CodeMirror initialization

intoeetive committed:
advanced configuration for CkEditor RTE

intoeetive committed:
support RTL text direction in Rich Text fields

Yulyaswan committed:
fixed tab layout code

intoeetive committed:
enable loading plugin javascript for RTE

Channel Form styling fixes for Redactor

intoeetive committed:
add CKEditor source to the main repo

intoeetive committed:
clear RTE Pages cache when Structure page is modified

Yulyaswan committed:
fixed sorting in new tab

intoeetive committed:
fix loading Redactor settings

intoeetive committed:
styling updates

intoeetive committed:
Resolved #2338 where vertical layout was not fully supported by Grid fields in Pro Variables

Yulyaswan committed:
fixed empty state styles for grid and pro var grid

Yulyaswan committed:
fixed view of bulk action for pro var list

bryannielsen committed:
Throw exception when createTempFile() fails

intoeetive committed:
fix the build script

intoeetive committed:
Resolved #2340 where Pro Variables of Grid type could show PHP notice on frontend

intoeetive committed:
Added "Upload Directory" column for File Manager table; #98

intoeetive committed:
Merge pull request #2341 from ExpressionEngine/bug/7.x/pro-vars-vertical-grid

Pro Variables fixes

intoeetive committed:
Merge pull request #2317 from ExpressionEngine/bug/7.x/layout-bug-EECORE-2094

Resolved #2131, #913 where moving fields between tabs in Channel Layouts did not work well

matthewjohns0n committed:
Merge pull request #2291 from ExpressionEngine/bug/7.x/filesystem-theme-path

Update default filesystem to use root of system path

intoeetive committed:
typo fix

intoeetive committed:
Resolved #1651 where dropdowns in File field were showing the directories that member was not allowed to access

intoeetive committed:
Enabled entering HTML line break tags into Entry Title fiels; #1633

intoeetive committed:
If the field belongs to group, redirect to group fields list after saving

intoeetive committed:
Revert "Resolved #1901 where add-ons with related models were not properly uninstalled"

This reverts commit 7e100bcfdf6494be1b7033e242649474817d83bc.

intoeetive committed:
Fixed the issue where empty edit buttons were present for sidebar elements that member does not have access to edit

intoeetive committed:
Fixed the issue where File link was not in navigation sidebar for some members with proper permissions

intoeetive committed:
Added "ungrouped" as a filter to the list of Fields; #516

matthewjohns0n committed:
Merge pull request #2325 from ExpressionEngine/bug/7.x/colorpicker-init-in-grid

Resolved #2279 where ColorPicker dropdown was not visible in new Grid rows

intoeetive committed:
Merge pull request #2322 from ExpressionEngine/bug/7.x/mfa-login-redirect-loop

Resolved #2116 where disabling MFA globally could cause redirect loop on login for some users

matthewjohns0n committed:
Merge pull request #2346 from ExpressionEngine/bug/7.x/addon-uninstall-revert

Revert "Resolved #1901 where add-ons with related models were not properly uninstalled"

intoeetive committed:
update relationship tests

intoeetive committed:
Show in which channels field is used on field edit page; #919

EECORE-2120

bakin1999 committed:
php 8+ warnings

bryannielsen committed:
Update local adapter to not require readable root

Yulyaswan committed:
fixed anchor name attr inside grid

intoeetive commented:
@mithra62 could you please also have this code PR'ed against `7.dev` in addition to this PR? I think you should be able to just do that from the same branch

intoeetive committed:
Merge branch '7.dev' into bug/7.x/parse-categories-in-related-categories-mode

eric-lamb62 commented:
Done @intoeetive 

https://github.com/ExpressionEngine/ExpressionEngine/pull/2367

intoeetive committed:
Update EE version to 7.1.4

intoeetive committed:
Merge pull request #2281 from ExpressionEngine/bug/7.x/parse-categories-in-related-categories-mode

Resolved #2135 where categories were not parsed in related categories mode

intoeetive committed:
Merge pull request #2275 from ExpressionEngine/bug/7.x/command-click-trigger-EECORE-2083

Prevent show confirm exit modal if cmd + click are pressed together

Yulyaswan committed:
added function to check grid width after save data or load page

matthewjohns0n committed:
Merge pull request #2289 from ExpressionEngine/bug/7.x/pre-populate-channel-field

Resolved #2285 where exception shown after creating select field populating from other field, but no source field selected

matthewjohns0n committed:
Merge pull request #2282 from ExpressionEngine/bug/7.x/preview-entry-grid-search

Resolved #2122 where Live Preview was not working correctly when template was using search on checkboxes column in Grid

intoeetive committed:
Resolved #2366 where is was not possible to use title in `search` parameter of `exp:channel:entries` tag

intoeetive committed:
make "require 2fa" toggle not conditional

igreil committed:
changed hardcoded "of" to proper language stringe

intoeetive committed:
fix getting correct mtime for some js files in combo loader

intoeetive committed:
Added extra confirmation toggle when deleting fieldtype that is used by fields; #484

EECORE-2128

Yulyaswan committed:
added edit entry button to relationship items

bryannielsen committed:
Update fallback filesystem to respect open_basedir

Yulyaswan committed:
fixed issue when file uploaded multiple times in File Grid

igreil committed:
Update Performance.php

bryannielsen committed:
Limit eager loading filesystem content to listings

bryannielsen committed:
Remove filesystem eager load from entity model

intoeetive committed:
Resolved issue where File Manager page was not updated after uploading SVG file

intoeetive committed:
file component fixes

Yulyaswan committed:
channel layout fixes

intoeetive committed:
reduced number of  `exists` checks on files

improve eager loading directory contents

impove eager loading snippets / variables

fix thumb generation on Windows when path is out of web root

intoeetive committed:
be less strict when eager loading directories

intoeetive committed:
copy-paste mistake, fixed

Yulyaswan committed:
fixed JavaScript error FileField is not defined  when uploading a file

robinsowell committed:
Fixed a PHP 8.1 issue in Structure add-on

Deprecated Automatic conversion of false to array is deprecated ee/ExpressionEngine/Addons/structure/sql.structure.php, line 1188
Severity: E_DEPRECATED

intoeetive committed:
Resolved #2393 where replacing file with existing manipulations could generate PHP error

robinsowell committed:
Fixed a PHP 8 bug when saving some member profile settings

If you have no checkboxes selected:

in_array(): Argument #2 ($haystack) must be of type array, string given

ee/ExpressionEngine/Controller/Members/Profile/Profile.php:325

intoeetive committed:
Resolved issue where entries could be not displayed on front-end if channel name also existed on other MSM site

EECORE-2133

intoeetive committed:
update version to 7.1.6

Yulyaswan committed:
Fixed Field_group test

intoeetive committed:
Resolved issue where PHP error could be shown after update if template are not synchronized

intoeetive committed:
fix conditional check for extension

robinsowell committed:
Fixed a bug where channel entry search parameter could only be used once

You should be able to do:

{exp:channel:entries channel="searching" search:seo_title="seriouslies|sblarg" search:seo_title="second|booter" search:contact_email="Third" search:contact_phone="first"}

But in the query, it was only querying one of the custom field tables instead of all 3.

intoeetive committed:
unset search column name after use, so that search would work properly

intoeetive committed:
Resolved #2395 where deleting directory might not work correctly on some environments

intoeetive committed:
enable parsing linebreaks for title on frontent

intoeetive committed:
PATH_JAVASCRIPT_BUILD should always point to build, not src

Yulyaswan committed:
Removed console.log

Yulyaswan committed:
fixed reloading page if entry editing from modal window

Yulyaswan committed:
Limited clickable area to input or text for checkboxes and radio buttons

Yulyaswan committed:
fixed tests

Yulyaswan committed:
Prevent hiding Redactor toolbar behind entry tabs when sticked to the top of the screen

intoeetive committed:
Resolved #2410 where some fieldtypes could not be used as columns in Entry Manager

matthewjohns0n committed:
Namespaced the Structure core nav parser to avoid conflicts with the EEHarbor codebase

intoeetive committed:
Resolved #2340 where editing template by member without settings & access permission was resetting allowed roles

EECORE-2145

eric-lamb62 commented:
Hey @intoeetive, 

Sorry for delay but I was able to create a PR for the docs for this new feature. 

https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/503 

Please lemme know if there's anything else required.

Yulyaswan committed:
added dataType json to new function

intoeetive committed:
Resolved #2431 where Image HTML button should not have been editable

EECORE-2144

Yulyaswan committed:
fixed code to display image in correct place for textarea

Yulyaswan committed:
Show edit button only if reationship isn't in a modal window

intoeetive committed:
Resolved #2438 when Low Variable types were not migrated properly to Pro Variable types

intoeetive committed:
ensuse modal context is passed to Relationship field, even in Fluid

intoeetive committed:
run tests using Chrome beta (#2445)

* run tests using Chrome beta

* chrome:beta is not available, use canary

* install Canary for tests

* canary not supported by platform, run beta

* run without beta flag

* some tests are still not updated properly, should remain skipped

intoeetive committed:
Merge pull request #2339 from ExpressionEngine/feature/7.x/rte-updates

RTE updates

intoeetive committed:
Merge pull request #2345 from ExpressionEngine/feature/7.x/br-wbr-in-entry-title

Enabled entering HTML line break tags into Entry Title fiels; #1633

intoeetive committed:
Merge pull request #2358 from ExpressionEngine/feature/display-revision-info-on-entry-edit

Added visual indication when viewing entry version; #1897

intoeetive committed:
Merge pull request #2419 from ExpressionEngine/bug/7.x/large-checkbox-area-EECORE-2140

Resolved #2403 where clickable area for checkboxes was too large

intoeetive committed:
Merge pull request #2378 from ExpressionEngine/feature/7.x/extra-delete-confirmation-for-fieldtypes-that-have-fields

Added extra confirmation toggle when deleting fieldtype that is used by fields; #484

intoeetive committed:
Merge branch '7.dev' into feature/7.x/edit-relationship-page-from-entry-EECORE-2116

# Conflicts:
#	themes/ee/asset/javascript/src/fields/rte/redactor/redactor.min.js
#	themes/ee/asset/javascript/src/fields/rte/redactor/redactor.min.js.map

intoeetive committed:
added EE 6.4.0 update file

intoeetive committed:
update controllers list for test

Yulyaswan committed:
fixed test

matthewjohns0n committed:
Disallow phar uploads

Yulyaswan committed:
resolved js conflict

intoeetive committed:
Merge pull request #2421 from ExpressionEngine/bug/7.x/redactor-toolbar-issue-EECORE-2141

Resolved #2361 where Redactor toolbar was hidden behind entry tabs when sticked to the top of the screen

intoeetive committed:
Update the license mentions of Pro components

intoeetive committed:
Merge pull request #2380 from ExpressionEngine/feature/7.x/edit-relationship-page-from-entry-EECORE-2116

Added ability to edit related entries content from relationship field; #656

matthewjohns0n committed:
Merge pull request #2456 from ExpressionEngine/feature/7.x/license-update

Update the license mentions of Pro components

matthewjohns0n committed:
Merge pull request #2444 from ExpressionEngine/bug/7.x/low-vars-to-pro-vars-type-update

Resolved #2438 when Low Variable types were not migrated properly to Pro Variable types

matthewjohns0n committed:
Merge branch '7.dev' into feature/6.x/object-addons

intoeetive commented:
Closing in favor of https://github.com/ExpressionEngine/ExpressionEngine/pull/2459 because we messed up here (
